SQF/dayz_code/actions/doorManagement/doorAddFriend.sqf

```diff
@@ -1,17 +1,24 @@
-private ["_pos","_plots","_inList","_friends","_toAdd"];
+private ["_inList","_toAdd","_friends","_friendUID","_friendName","_userList"];
+disableSerialization;
+
+_userList = (findDisplay 711195) displayCtrl 7101;
+
+_friendUID = _this select 0;
+_friendName = _userList lbText (lbCurSel _userList);
+
+if (_friendUID == "") exitWith {};
 
-_pos = _this select 0;
-if (_pos < 0) exitWith {};
-_toAdd = (Humans select _pos);
 _friends = TheDoor getVariable ["doorfriends",[]];
+_toAdd = [_friendUID,toArray _friendName];
 _inList = false;
-{
-	if((_x  select 0) == (_toAdd select 0)) exitWith { _inList = true; };
-} forEach _friends;
-if(_inList) exitWith { localize "STR_EPOCH_PLOTMANAGEMENT_ADDFRIEND_ALREADYONTHELIST" call dayz_rollingMessages; };
-if(count _friends >= DZE_doorManagementMaxFriends) exitWith { format[localize "STR_EPOCH_PLOTMANAGEMENT_ADDFRIEND_FRIENDLIMIT", DZE_doorManagementMaxFriends] call dayz_rollingMessages; };
-_friends set [count _friends, _toAdd];
+
+{ if ((_x  select 0) == (_toAdd select 0)) exitWith { _inList = true; }; } count _friends;
```

Comment:
It is possible the exitWith will not run. In that case this will not return a bool and will trigger a "Type x expected bool" RPT error. To fix that just add a bool at the end of the count loop like so:
```sqf
{ if ((_x  select 0) == (_toAdd select 0)) exitWith { _inList = true; }; false } count _friends;
```